### PR TITLE
Refactoring and fixing type annotations in rule.py

### LINKED ISSUE #6454: Type annotate the code
### Description

Type annotate methods which have not been annotated yet.
Establish common types. Possibly re-write methods to conform to these common types.

### Motivation

_No response_

### Change

_No response_

### PR DESCRIPTION
**Pull request by rdimaio**
Addressing changes raised here https://github.com/rucio/rucio/pull/6516

#6454

### CODE REVIEW

File: lib/rucio/core/rule.py [hunk 1]
```diff
@@ -1856,7 +1865,7 @@ def get_injected_rules(total_workers: int,
                        blocked_rules: Sequence[str] = [],
                        *,
                        session: "Session"
-                       ) -> list[tuple[str, str]]:
+                       ) -> list[str]:
```
Comment (dchristidis) [inline lib/rucio/core/rule.py]: I’m afraid I might have not made myself clear in https://github.com/rucio/rucio/pull/6516#discussion_r1523231979.

The current return type *is* `list[tuple[str]]`.

You can see it from:

https://github.com/rucio/rucio/blob/ec84fe0f2a9aefc429448778ba5d4f63a6cbe94b/lib/rucio/core/rule.py#L1844

`rule[0]` is just the rule ID, then `rule` is a single-element tuple. The Injector handles it as:

https://github.com/rucio/rucio/blob/ec84fe0f2a9aefc429448778ba5d4f63a6cbe94b/lib/rucio/daemons/judge/injector.py#L78-L90

But there is no benefit to doing this! So my request was to take the opportunity to 1) *change* `get_injected_rules()` to return `list[str]` and then 2) *adjust* the Injector to handle it. Same for `get_stuck_rules()` and the Repairer.

So, unfortunately, the correction to the hints in this PR is wrong again.

**Review by dchristidis — DISMISSED** [review]
Under different circumstances I would have requested further changes. However, given the urgency to prepare the next release candidate, I believe it’s fine to merge it in its current state.

File: lib/rucio/core/rule.py (hunk 1 above)
Comment (rdimaio) [reply lib/rucio/core/rule.py]: Addressed in latest revision

**Comment by rdimaio** [comment]
Re-formatted to follow style outlined here https://github.com/rucio/rucio/pull/6516#discussion_r1523140076

**Review by dchristidis — APPROVED** [review]

**Pull request merged**